src/actions/app.js

```diff
@@ -168,19 +168,30 @@ export const { hideAlert } = createActions({
 
 export const { runQuery } = createActions({
   [ActionTypes.RUN_QUERY_SUCCESS]: (result) => result,
-  [ActionTypes.RUN_QUERY]: (projectId, query, maxResults) => ({
+  [ActionTypes.RUN_QUERY]: (projectId, query) => ({
     projectId,
     query,
-    maxResults,
+  }),
+});
+
+export const { previewData } = createActions({
+  [ActionTypes.PREVIEW_DATA_SUCCESS]: (queryResults, columns) => ({
```

Comment:
minor: could we swap the order?  I like `FOO_SUCCESS` being after `FOO`

Reply:
should we also have a FAILURE handler?

Reply:
Good point - added. I added a "Failure" component as a sorta placeholder -- probably should come up with a design for handling the case when the request fails. Right now it's just an error icon + error message.